synthesizer/process/src/lib.rs

```diff
@@ -66,30 +66,61 @@ use synthesizer_program::{
 };
 use synthesizer_snark::{ProvingKey, UniversalSRS, VerifyingKey};
 
-use aleo_std::prelude::{finish, lap, timer};
-use indexmap::IndexMap;
-use parking_lot::RwLock;
-use std::{collections::HashMap, sync::Arc};
+use aleo_std::{
+    StorageMode,
+    prelude::{finish, lap, timer},
+};
+use lru::LruCache;
+use parking_lot::{Mutex, RwLock};
+use std::{collections::HashMap, num::NonZeroUsize, sync::Arc};
+use tracing::debug;
 
 #[cfg(feature = "aleo-cli")]
 use colored::Colorize;
 
+#[cfg(not(feature = "rocks"))]
+use ledger_store::helpers::memory::ConsensusMemory;
+#[cfg(feature = "rocks")]
+use ledger_store::helpers::rocksdb::ConsensusDB;
+
+#[cfg(feature = "rocks")]
+#[derive(Clone)]
+pub struct Process<N: Network> {
+    /// The universal SRS.
+    universal_srs: Arc<UniversalSRS<N>>,
+    /// The Stack for credits.aleo
+    credits: Option<Arc<Stack<N>>>,
+    /// The mapping of program IDs to stacks.
+    stacks: Arc<Mutex<LruCache<ProgramID<N>, Arc<Stack<N>>>>>,
+    /// The storage.
+    store: Option<ConsensusStore<N, ConsensusDB<N>>>,
+}
+
+#[cfg(not(feature = "rocks"))]
 #[derive(Clone)]
 pub struct Process<N: Network> {
     /// The universal SRS.
     universal_srs: Arc<UniversalSRS<N>>,
```

Comment:
while we're at it, the `UniversalSRS` doesn't need to be in an `Arc`, as it's already arced internally and derives `Clone`
```suggestion
    universal_srs: UniversalSRS<N>,
```

Reply:
This breaks at least one other place in the code which assumes this is an Arc, probably trivial to fix but can be done outside of this PR.